### ISSUE TITLE
Evaluation detector YOLOv12

### ISSUE THREAD

**Issue by carmecorbi** [ultralytics/ultralytics#20786]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have my dataset annotated with two classes: 0 for person and 1 for sports ball.
In my data.yaml file, I have defined:

path: /data-fast/data-server/ccorbi/dataset_tracking/full
train: images/train
val: images/val
test: images/test

names:
    0: person
    1: sports ball

However, when I evaluate on the validation dataset, class 1 is evaluated as bicycle (which is class 1 in the COCO classes). In COCO, sports ball corresponds to class 32.

So my question is: how should I label my data if I only want to detect the classes person and sports ball?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @carmecorbi, thank you for reaching out and using Ultralytics 🚀! This is an automated response to help guide you through the resources and next steps while an Ultralytics engineer reviews your question.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for details on dataset formatting and class labeling, as well as many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Common questions about custom datasets and class mappings are often answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug further.

If this is a custom training ❓ Question (such as label configuration), please include as much detail as possible—such as example images, your exact `data.yaml`, and relevant logs. Also, reviewing our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) can help ensure your setup is optimal.

To connect with the Ultralytics community:
- For real-time chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, visit [Discourse](https://community.ultralytics.com/)
- Or share and learn with others on the [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), as recent updates may resolve your issue:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing and verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your question and assist you soon. Thank you for your patience and for being part of the Ultralytics community!

**Comment by Y-T-G** [comment]
You will have to train the model first on your dataset and then use that model for evaluation.

**Comment by carmecorbi** [comment]
Thank you for the response!

I understand that to get consistent results with my labels, I need to train the model on my own dataset. However, what I would like to do first is evaluate the pre-trained YOLO model (such as yolov8n.pt) without training it, just by running inference and evaluation — but only on two specific COCO classes: person (class 0) and sports ball (class 32).

Is there a way to do this? Since the pre-trained model already knows all COCO classes, I would like to perform inference and evaluation while filtering for just these two classes.
